custom_components/zwave_mqtt/binary_sensor.py

```diff
@@ -110,56 +343,40 @@ def entity_registry_enabled_default(self) -> bool:
         return False
 
 
-class ZWaveListSensor(ZWaveDeviceEntity, BinarySensorDevice):
-    """Representation of a ZWaveListSensor translated to binary_sensor."""
-
-    @property
-    def is_on(self):
-        """Return if the sensor is on or off."""
-        return self.values.primary.value["Selected"] != "Clear"
-
-    @property
-    def device_state_attributes(self):
-        """Return the device specific state attributes."""
-        attributes = super().device_state_attributes
-        attributes["event"] = self.values.primary.value["Selected"]
-        return attributes
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return DEVICE_CLASS_MAPPING.get(self.values.primary.index)
-
-    @property
-    def entity_registry_enabled_default(self) -> bool:
-        """Return if the entity should be enabled when first added to the entity registry."""
-        # We hide some of the more advanced sensors by default to not overwhelm users
-        if self.values.primary.index in [8, 9]:
-            return False
-        return True
-
-
 class ZWaveListValueSensor(ZWaveDeviceEntity, BinarySensorDevice):
     """Representation of a ZWaveListValueSensor binary_sensor."""
 
-    def __init__(self, values, list_value, device_class=None):
-        """Initilize a ZWaveListValueSensor entity."""
+    def __init__(self, values, list_value, device_class=None, default_enabled=True):
+        """Initialize a ZWaveListValueSensor entity."""
         self._list_value = list_value
         self._device_class = device_class
+        self._default_enabled = default_enabled
+        self._state = False
         super().__init__(values)
 
+    @callback
+    def values_updated(self):
+        """Call when the underlying values are added or updated."""
+        # Entity state only returns false if clear is reported
+        # So we do not directly map the value but wait for the value change
+        if self.values.primary.value["Selected_id"] == self._list_value:
+            self._state = True
+        elif self.values.primary.value["Selected_id"] == 0:
+            self._state = False
```

Comment:
Just to repeat what I said above, for most of these this is true, but not all, such as the battery status or heat alarm heat sensor status, that have more than one state. So this might be a little simplistic for some state variables.

Reply:
This is exactly what handles that. It will only reset if a clear event has been seen.

Reply:
Does this still work for doors then, or other sensors that don't support Idle (Clear)? A door/window open state is either 22 or 23, never 0. It follows this:
![image](https://user-images.githubusercontent.com/791794/80461585-5c456d00-88ea-11ea-9bfc-245f9775bfae.png)

Reply:
it's unclear to me. What I understand is that a "previous event cleared" will be sent on the transition. Let's wait for a reply by @Fishwaldo about this.

If the clear event is nog sent, the code is going to be a lot more complex, as we need to handle each specific situation.

Reply:
That is not always true, such as the example I just posted. Not all variables support idle.

Reply:
If you have not already, take a look at [SDS13713 Notification Command Class, list of assigned Notifications](https://www.silabs.com/documents/login/miscellaneous/SDS13713-Notification-Command-Class.xlsx).

Access control "door state" state variable is defined as having the values 0x16 (open) or 0x17 (closed), and Idle is not applicable.

![image](https://user-images.githubusercontent.com/791794/80468459-0c6ba380-88f4-11ea-9b7d-a57210611de1.png)

Reply:
Clear is not meant as idle but as event to indicate the previous event cleared. So if we spot the event "door open" we set the binary_sensor ON. Only when "clear comes along we set it to OFF again.
My assumption is that OZW sends "door open" --> "clear" --> "door closed"

Reply:
The screen shots I've been posting are from the spec. From [OZW](https://github.com/OpenZWave/open-zwave/blob/5220be772970b922fd6304f9d347d025a6511e7a/config/NotificationCCTypes.xml#L154-L157):
```xml
	<AlarmType id="6" name="Access Control">
		<AlarmEvent id="0" name="Clear">
			<AlarmEventParam id="256" type="byte" name="Previous Event Cleared" />
		</AlarmEvent>
```
"Clear" is what OZW calls the Idle State. The value is 0x00 and the event parameter is the value of the variable that went Idle, aka "Previous Event Cleared", as shown in the screen shot.

As in the screen shot, the "Door state" variable is either open or closed. Idle is not applicable as indicated by "N/A".

Anyways, time for me to head off, I'm sure it will get sorted. 👍

Reply:
omg, that would be horrible as it will require specific code for each situation. 
Anyways, I might have an idea to fix it in a generic way.

Reply:
Thanks for your help on this

Reply:
Solved. We can now handle the situation that the sensor is reset as soon as a specific other value is reached (the doorsensor example), the situation that a value resets as soon as just another value is send and the more regular "value resets on clear"